### PR TITLE
Rateconvfixed

### PR DESCRIPTION
**Pull request by plgbrts**
Fix for the situation that hydrocarbon consist only of gas. Relevant for water-gas twophase system

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/simulators/wells/RateConverter.hpp [hunk 1]
```diff
@@ -496,20 +496,34 @@ namespace Opm {
...
-                        attr.rs += fs.Rs().value() * pv_cell;
-                        attr.rv += fs.Rv().value() * pv_cell;
-                        attr.temperature += fs.temperature(FluidSystem::oilPhaseIdx).value() * pv_cell;
+                        if (Details::PhaseUsed::oil(pu)) {
+                            attr.pressure += fs.pressure(FluidSystem::oilPhaseIdx).value() * pv_cell;
+                            attr.rs += fs.Rs().value() * pv_cell;
+                            attr.rv += fs.Rv().value() * pv_cell;
+                            attr.temperature += fs.temperature(FluidSystem::oilPhaseIdx).value() * pv_cell;
+                        }
+                        else{
```
Comment (GitPaean) [inline opm/simulators/wells/RateConverter.hpp:522]: very nitpicking here, please do `} else {` to make it consistent with other code in the same file.

File: opm/simulators/wells/RateConverter.hpp [hunk 2]
```diff
@@ -496,20 +496,34 @@ namespace Opm {
...
                     if (pv_cell > 0.) {
                         auto& attr = attributes_pv[reg];
                         attr.pv += pv_cell;
-                        attr.pressure += fs.pressure(FluidSystem::oilPhaseIdx).value() * pv_cell;
-                        attr.rs += fs.Rs().value() * pv_cell;
-                        attr.rv += fs.Rv().value() * pv_cell;
-                        attr.temperature += fs.temperature(FluidSystem::oilPhaseIdx).value() * pv_cell;
+                        if (Details::PhaseUsed::oil(pu)) {
+                            attr.pressure += fs.pressure(FluidSystem::oilPhaseIdx).value() * pv_cell;
+                            attr.rs += fs.Rs().value() * pv_cell;
```
Comment (GitPaean) [inline opm/simulators/wells/RateConverter.hpp:518]: To be consistent for the situation that only has either oil or gas as the only hydrocarbon component,  you can either move 
```c++                         
attr.rs += fs.Rs().value() * hydrocarbonPV;
attr.rv += fs.Rv().value() * hydrocarbonPV; 
```
out of the `if` statement or you can put them inside an `if` statement like the following,
```c++
if (Details::PhaseUsed::oil(pu) && Details::PhaseUsed::gas(pu))
```

File: opm/simulators/wells/RateConverter.hpp [hunk 3]
```diff
@@ -496,20 +496,34 @@ namespace Opm {
...
                         auto& attr = attributes_hpv[reg];
                         attr.pv += hydrocarbonPV;
-                        attr.pressure += fs.pressure(FluidSystem::oilPhaseIdx).value() * hydrocarbonPV;
-                        attr.rs += fs.Rs().value() * hydrocarbonPV;
-                        attr.rv += fs.Rv().value() * hydrocarbonPV;
-                        attr.temperature += fs.temperature(FluidSystem::oilPhaseIdx).value() * hydrocarbonPV;
+                        if (Details::PhaseUsed::oil(pu)) {
+                            attr.pressure += fs.pressure(FluidSystem::oilPhaseIdx).value() * hydrocarbonPV;
+                            attr.rs += fs.Rs().value() * hydrocarbonPV;
+                            attr.rv += fs.Rv().value() * hydrocarbonPV;
```
Comment (GitPaean) [inline opm/simulators/wells/RateConverter.hpp:502]: Same comment here as the comments below.

**Review by GitPaean — CHANGES_REQUESTED** [review]
Thank you for the contributions. There is some nitpicking, otherwise, it looks good to me.

File: opm/simulators/wells/RateConverter.hpp (hunk 3 above)
Comment (plgbrts) [reply opm/simulators/wells/RateConverter.hpp:501]: Followed your suggestion.

File: opm/simulators/wells/RateConverter.hpp (hunk 2 above)
Comment (plgbrts) [reply opm/simulators/wells/RateConverter.hpp:518]: Followed your suggestion to put it in a separate if statement. Thanks for reviewing!

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by plgbrts** [comment]
@GitPaean it is not clear to me why certain test fails. 
The error message is 
_/var/lib/jenkins/workspace/opm-simulators-PR-builder/tests/run-regressionTest.sh: line 19: 25103 Illegal instruction     ${BINPATH}/${EXE_NAME} ${TEST_ARGS} --output-dir=${RESULT_PATH}_
Do you know the cause of this?

**Comment by joakim-hove** [comment]
> [....,] it is not clear to me why certain test fails.

I think it is a jenkins/build-servee problem, not related to your cose.

**Comment by GitPaean** [comment]
> I think it is a jenkins/build-servee problem, not related to your cose.

I think it is probably correct. I could not reproduce the problem locally.  

The PR might cause some failure related to RS RV output, which remains to be verified. 

@akva2 , any suggestions here?

**Comment by akva2** [comment]
not entirely sure what was up, but the box has been rebooted and i think everything is back in order.

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
all failures now looks sane as in they are not caused by invalid opcode segfaults.

**Comment by akva2** [comment]
jenkins build this opm-tests=573 please

**Comment by akva2** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
@akva2 , thanks for fixing the test. Now everything looks fine. I am getting this PR in now.

**Review by GitPaean — APPROVED** [review]

**Pull request merged**